v2/coord/conversion.go

```diff
@@ -0,0 +1,57 @@
+package coord
+
+import (
+	"github.com/libp2p/go-libp2p/core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/plprobelab/go-kademlia/kad"
+
+	"github.com/libp2p/go-libp2p-kad-dht/v2/kadt"
+)
+
+// NodeInfoToAddrInfo converts a kad.NodeInfo to a peer.AddrInfo.
+// This function will panic if info.ID() does not return a kadt.PeerID
+func NodeInfoToAddrInfo(info kad.NodeInfo[KadKey, ma.Multiaddr]) peer.AddrInfo {
+	peerID := info.ID().(kadt.PeerID)
+	return peer.AddrInfo{
+		ID:    peer.ID(peerID),
+		Addrs: info.Addresses(),
+	}
+}
+
+// NodeIDToAddrInfo converts a kad.NodeID to a peer.AddrInfo with no addresses.
+// This function will panic if id's underlying type is not kadt.PeerID
+func NodeIDToAddrInfo(id kad.NodeID[KadKey]) peer.AddrInfo {
+	peerID := id.(kadt.PeerID)
+	return peer.AddrInfo{
+		ID: peer.ID(peerID),
```

Comment:
could use `NodeIDToPeerID`

Reply:
Ideally we remove these ugly helpers. They are here to ease the transition between the original generic code and the non-generic form.